nixos/modules/services/networking/wireguard.nix

```diff
@@ -153,35 +182,43 @@ let
   wgCommand = "${pkgs.wireguard}/bin/wg";
 
   generateUnit = name: values:
+    assert (values.privateKey != null || values.privateKeyFile != null);
     nameValuePair "wireguard-${name}"
       {
         description = "WireGuard Tunnel - ${name}";
         after = [ "network.target" ];
         wantedBy = [ "multi-user.target" ];
+        script = builtins.concatStringsSep "\n" (lib.flatten([
+          values.preSetup
+
+          ''
+          ${ipCommand} link del dev "${name}" || true
+          ${ipCommand} link add dev ${name} type wireguard
+          ${wgCommand} setconf ${name} /dev/stdin <<EOF
```

Comment:
It doesn't make a difference in terms of WireGuard.

It might make a difference in terms of argv stack space though if you have thousands of peers. So if you go the wg set route, I'd recommend you first set up the device:

```
wg set wg0 private-key ... listen-port ...
```

And then set up each peer separately:

```
wg set wg0 peer ABCDEF= endpoint ... preshared-key ...
wg set wg0 peer FAA1234= endpoint ... preshared-key ...
...
```

Also, keep in mind that @Mic92's suggestion of <(echo whatever) requires bash and won't work with generic sh.